packages/devtools_app/lib/src/code_size/code_size_table.dart

```diff
@@ -0,0 +1,129 @@
+import 'package:flutter/material.dart';
+
+import '../charts/treemap.dart';
+import '../table.dart';
+import '../table_data.dart';
+import '../utils.dart';
+
+class CodeSizeTable extends StatelessWidget {
+  factory CodeSizeTable({
+    @required rootNode,
+    @required totalSize,
+  }) {
+    final treeColumn = _LibraryRefColumn(currentRootLevel: rootNode.level);
+    final sortColumn = _SizeColumn();
+    final columns = List<ColumnData<TreemapNode>>.unmodifiable([
+      treeColumn,
+      sortColumn,
+      _SizePercentageColumn(totalSize: totalSize),
+    ]);
+    return CodeSizeTable._(
+      rootNode,
+      totalSize,
+      treeColumn,
+      sortColumn,
+      columns,
+    );
+  }
+
+  const CodeSizeTable._(
+    this.rootNode,
+    this.totalSize,
+    this.treeColumn,
+    this.sortColumn,
+    this.columns,
+  );
+
+  final TreemapNode rootNode;
+  final int totalSize;
+
+  final TreeColumnData<TreemapNode> treeColumn;
+  final ColumnData<TreemapNode> sortColumn;
+  final List<ColumnData<TreemapNode>> columns;
+
+  @override
+  Widget build(BuildContext context) {
+    return TreeTable<TreemapNode>(
+      dataRoots: rootNode.children,
+      columns: columns,
+      treeColumn: treeColumn,
+      keyFactory: (node) => PageStorageKey<String>(node.name),
+      sortColumn: sortColumn,
```

Comment:
Would it be okay to write `treeColumn: columns[0]` and `sortColumn: columns[1]` instead if they are the same thing? Doing so would remove about 15 lines above although we would lose some immediate clarity in what they are referring to.

Reply:
it's generally not a good idea to reference an index. If the order of the `columns` in the table were to change at some point, referencing an index here would break the code. This code is still very clean and concise as is

Reply:
Sounds good